app/models/bookmark.rb

```diff
@@ -4,6 +4,7 @@ class Bookmark < ActiveRecord::Base
   belongs_to :user
   belongs_to :post
   has_one :topic, through: :post
+  belongs_to :bookmarkable, polymorphic: true
```

Comment:
I think it'll be helpful as well to declare the association on the `Post` and `Topic` model.

Reply:
Sure, I will add that in.

Reply:
Actually, no I can't add that in, because those models already have associations based on the old columns. I can change that over in the PR where I ignore the old columns:

https://github.com/discourse/discourse/blob/62f393b933902f4b101cd9b081f7654e602b5401/app/models/topic.rb#L206-L206

https://github.com/discourse/discourse/blob/62f393b933902f4b101cd9b081f7654e602b5401/app/models/post.rb#L49-L49

Reply:
Update: These relationships actually point to the polymorphic columns by default so I've had to replace them with a method for now.